### PR TITLE
Skr svcat install example resources

### PR DESCRIPTION
**Pull request by voigt**
**Description**

Changes proposed in this pull request:

Deploys the following ressources as part of the BTP migration test:

```yaml
kubectl get serviceinstance,servicebinding,servicebindingusage,function | grep -v "service-catalog-auditlog-api"
NAME                                                                 CLASS                                     PLAN          STATUS             AGE
serviceinstance.servicecatalog.k8s.io/hb-instbind-redis-1            ClusterServiceClass/redis                 micro         Ready              2d20h
serviceinstance.servicecatalog.k8s.io/svcat-auditlog-api-1           ClusterServiceClass/auditlog-api          default       Ready              2d20h
serviceinstance.servicecatalog.k8s.io/svcat-auditlog-management-1    ClusterServiceClass/auditlog-management   default       Ready              2d20h
serviceinstance.servicecatalog.k8s.io/svcat-html5-apps-repo-1        ClusterServiceClass/html5-apps-repo       app-runtime   Ready              2d20h

NAME                                                                       SERVICE-INSTANCE              SECRET-NAME                           STATUS   AGE
servicebinding.servicecatalog.k8s.io/func-sb-redis-function-1              hb-instbind-redis-1           func-sb-redis-function-1              Ready    2d20h
servicebinding.servicecatalog.k8s.io/func-sb-svcat-auditlog-api-1          svcat-auditlog-api-1          func-sb-svcat-auditlog-api-1          Ready    2d20h
servicebinding.servicecatalog.k8s.io/func-sb-svcat-auditlog-management-1   svcat-auditlog-management-1   func-sb-svcat-auditlog-management-1   Ready    2d20h
servicebinding.servicecatalog.k8s.io/func-sb-svcat-html5-apps-repo-1       svcat-html5-apps-repo-1       func-sb-svcat-html5-apps-repo-1       Ready    2d20h
servicebinding.servicecatalog.k8s.io/inst-sb-redis-function-1              hb-instbind-redis-1           inst-sb-redis-function-1              Ready    2d20h
servicebinding.servicecatalog.k8s.io/inst-sb-svcat-auditlog-api-1          svcat-auditlog-api-1          inst-sb-svcat-auditlog-api-1          Ready    2d20h
servicebinding.servicecatalog.k8s.io/inst-sb-svcat-auditlog-management-1   svcat-auditlog-management-1   inst-sb-svcat-auditlog-management-1   Ready    2d20h
servicebinding.servicecatalog.k8s.io/inst-sb-svcat-html5-apps-repo-1       svcat-html5-apps-repo-1       inst-sb-svcat-html5-apps-repo-1       Ready    2d20h

NAME                                                                                      AGE
servicebindingusage.servicecatalog.kyma-project.io/func-sbu-svcat-auditlog-api-1          2d20h
servicebindingusage.servicecatalog.kyma-project.io/func-sbu-svcat-auditlog-management-1   2d20h
servicebindingusage.servicecatalog.kyma-project.io/func-sbu-svcat-html5-apps-repo-1       2d20h
servicebindingusage.servicecatalog.kyma-project.io/hb-instbind-redis-1                    2d20h

NAME                                                              CONFIGURED   BUILT   RUNNING   RUNTIME    VERSION   AGE
function.serverless.kyma-project.io/hb-instbind-redis-1           True         True    True      nodejs12   4         2d20h
function.serverless.kyma-project.io/svcat-auditlog-api-1          True         True    True      nodejs12   4         2d20h
function.serverless.kyma-project.io/svcat-auditlog-management-1   True         True    True      nodejs12   4         2d20h
function.serverless.kyma-project.io/svcat-html5-apps-repo-1       True         True    True      nodejs12   4         2d20h
```

**Related issue(s)**
prow job: kyma-project/test-infra#3913
epic: https://github.tools.sap/kyma/backlog/issues/1656

### CODE REVIEW

File: tests/fast-integration/skr-svcat-migration-test/deploy-sample-resources.js [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+    waitForPodWithLabel,
+    waitForFunction,
+    waitForServiceInstance,
+    waitForServiceBinding,
+    waitForServiceBindingUsage,
+    waitForClusterAddonsConfiguration,
+    getClusteraddonsconfigurations,
+  } = require("../utils");
+
+async function installRessource(manifest, ressource, name, namespace) {
```
Comment (wozniakjan) [inline tests/fast-integration/skr-svcat-migration-test/deploy-sample-resources.js:13]: nit: consider `s/ressource/resource/`

File: tests/fast-integration/skr-svcat-migration-test/deploy-sample-resources.js [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+  
+  await installRessource(serviceInstanceManifestPath, "serviceInstance", "svcat-auditlog-management-1", "default")
+  await installRessource(functionManifestPath, "function", "svcat-auditlog-management-1", "default")
+  await installRessource(funcSBManifestPath, "servicebinding", "func-sb-svcat-auditlog-management-1", "default")
+  await installRessource(instSBManifestPath, "servicebinding", "inst-sb-svcat-auditlog-management-1", "default")
+  await installRessource(sbuManifestPath, "servicebindingusage", "func-sbu-svcat-auditlog-management-1", "default")
+  
+  // return console.timeEnd('installAuditManagementExample')
+  let t1 = performance.now()
+  console.log(`Finished deployment of Audit-Management in ${t1-t0} ms`)
```
Comment (wozniakjan) [inline tests/fast-integration/skr-svcat-migration-test/deploy-sample-resources.js:180]: this is pretty good idea, the testing framework does output elapsed time for each `It()` block that is over 5000ms.
```
  SKR SVCAT migration test
    ✓ Should provision SKR (1711998ms)
    ✓ Should save kubeconfig
    ✓ Should initialize K8s
    ✓ Should instantiate SM Instance and Binding (7749ms)
    ✓ Should install BTP Operator helm chart (22789ms)
    ✓ Should deprovision SKR (2371940ms)
    ✓ Should cleanup SM instances and bindings
```
But I like your approach and I agree we may want to have some more detailed and better consumable metrics to compose statistics than the default ms for slow `It()` blocks.

**Review by wozniakjan — COMMENTED** [review]
few minor nits and comments below

File: tests/fast-integration/skr-svcat-migration-test/deploy-sample-resources.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (voigt) [reply tests/fast-integration/skr-svcat-migration-test/deploy-sample-resources.js]: Thanks, done!

File: tests/fast-integration/utils/index.js [hunk 3]
```diff
@@ -245,6 +245,17 @@ async function getAllCRDs() {
 
 }
 
+async function getClusteraddonsconfigurations() {
+  const path = '/apis/addons.kyma-project.io/v1alpha1/clusteraddonsconfigurations';
+  const response = await k8sDynamicApi.requestPromise({
+    url: k8sDynamicApi.basePath + path
+  });
+  const stat = {}
```
Comment (pbochynski) [inline tests/fast-integration/utils/index.js:253]:
```suggestion
```

File: tests/fast-integration/utils/index.js [hunk 4]
```diff
@@ -245,6 +245,17 @@ async function getAllCRDs() {
...
+  const body = JSON.parse(response.body);
+  body.items.forEach(crd => stat[crd.spec.group] = stat[crd.spec.group] ? stat[crd.spec.group] + 1 : 1)
```
Comment (pbochynski) [inline tests/fast-integration/utils/index.js:255]:
```suggestion
```

File: tests/fast-integration/skr-svcat-migration-test/fixtures/02_serviceinstance_auditlog.yaml [hunk 5]
```diff
@@ -0,0 +1,10 @@
+---
+apiVersion: servicecatalog.k8s.io/v1beta1
+kind: ServiceInstance
+metadata:
+  name: svcat-auditlog-api-1
+  namespace: default
```
Comment (pbochynski) [inline tests/fast-integration/skr-svcat-migration-test/fixtures/02_serviceinstance_auditlog.yaml:6]: Remove namespace from yaml. You won't be able to install it in another namespace with current implementation of kubectlApply (namespace is applied only for resources where the namespace is not given).

**Review by pbochynski — CHANGES_REQUESTED** [review]
I would add your test case to the getting started guide test. I see you create the same types of resources (redis service instance, function, SBU, etc). 
You can save about 30-40 seconds required to build a function if you reuse getting started fixture.

File: tests/fast-integration/skr-svcat-migration-test/skr-svcat-migration-test.js [hunk 6]
```diff
@@ -49,6 +50,9 @@ describe("SKR SVCAT migration test", function() {
   it(`Should instantiate SM Instance and Binding`, async function() {
     btpOperatorCreds = await t.smInstanceBinding(smAdminCreds, svcatPlatform, btpOperatorInstance, btpOperatorBinding);
   });
+  it(`Should install sample service catalogue ressources`, async function() {
+    await deploy()
```
[the remark this answers is not in the document]
Comment (voigt) [reply tests/fast-integration/skr-svcat-migration-test/skr-svcat-migration-test.js]: Added a destroy() job to remove resources before SKR shutdown.

**Comment by voigt** [comment]
> I would add your test case to the getting started guide test. I see you create the same types of resources (redis service instance, function, SBU, etc).
> You can save about 30-40 seconds required to build a function if you reuse getting started fixture.

@pbochynski We will refactor this test once it is fully operational and prove that migration is working. This has been discussed with @PK85  as well.

**Review by PK85 — APPROVED** [review]

**Review by pbochynski — APPROVED** [review]
Conditional approval - please merge (remove) getting started guide with this test.

**Pull request merged**